lib/services/identity.go

```diff
@@ -51,6 +51,12 @@ type UsersService interface {
 	DeleteUser(ctx context.Context, user string) error
 	// GetUsers returns a list of users registered with the local auth server
 	GetUsers(withSecrets bool) ([]User, error)
+}
+
+// LocalUsers manages users on auth server
+type LocalUsers interface {
```

Comment:
can this be in `lib/auth`?
same for all `Local*` interfaces in this package

Reply:
All LocalXXX interfaces are supposed to represent the auth server's view. Currently I renamed them to ServerXXX and kept in lib/services but I could follow your suggestion to move these to lib/auth and shuffle the references around but wanted to make sure it's what you intended. 

I could keep the client-side interfaces with original names, otherwise it'd look like the following if I understood you correctly:
```go
[lib/auth]

type trust interface {
   // ...
}

type ClientTrust interface {
   trust
}

type ServerTrust interface {
   trust
   // additional APIs
}

in this setup, trust is implicitly a client-side interface so all exported client interface will just embed that without adding anything new.
```

Reply:
One example where client and server interfaces differ is streaming gRPC:
```
// server
AddMFADevice(stream proto.AuthService_AddMFADeviceServer) error
// client
AddMFADevice(ctx context.Context) (proto.AuthService_AddMFADeviceClient, error)
```

I think explicit `ClientXXX` wrappers (even though they would just embed `XXX` most of the time) would make code easier to understand. The current confused me a lot when I joined the team.

Reply:
Yes, I realized that there were already cases of that - needed some time to process that internally though. Thanks for the pointers.